### PR TITLE
Expand ROMs and Add/Remove headers from CLI

### PR DESCRIPTION
**Pull request by LittleCube13**
I had to do some weird things with the functions to get this to work, if you have any problems or ideas on how to more efficiently use the functions, don't hesitate to point them out.

### CODE REVIEW

File: coilsnake/ui/gui_util.py [hunk 1]
```diff
@@ -6,155 +6,171 @@
...
-        title="Select a Project Directory",
-        mustexist=(not save))
-    if filename:
-        set_entry_text(entry, filename)
-        entry.xview(END)
+	filename = tkFileDialog.askdirectory(
+		parent=root,
+		initialdir=os.path.dirname(entry.get()) or os.path.expanduser("~"),
+		title="Select a Project Directory",
+		mustexist=(not save))
```
Comment (mrtenda) [inline coilsnake/ui/gui_util.py]: please don't change the indenting of code you're not modifying

File: coilsnake/ui/cli.py [hunk 2]
```diff
@@ -45,6 +45,26 @@ def main():
     patch_rom_parser.add_argument("patch")
     patch_rom_parser.add_argument("headered")
     patch_rom_parser.set_defaults(func=_patchrom)
+    
+    expand32_parser = subparsers.add_parser("expand32", help="Expand a ROM's size to 32 MBits (4MB)")
```
Comment (mrtenda) [inline coilsnake/ui/cli.py]: would it be better if there was a single "expand" command that takes in the size as an additional parameter (32 or 48)?

**Comment by LittleCube13** [comment]
Yeah, sorry, I didn't realize that had happened with the indentation. I'll make sure to be more careful. And I'll also combine the expand functions.

**Comment by LittleCube13** [comment]
I pushed another commit, in this one i also reverted the gui_util.py indentation and some of the other indentation. again, sorry about that.

File: coilsnake/ui/common.py [hunk 3]
```diff
@@ -266,6 +266,39 @@ def patch_rom(clean_rom_filename, patched_rom_filename, patch_filename, headered
 
     log.info("Patched to {} in {:.2f}s".format(patched_rom_filename, time.time() - patching_start_time))
 
+def expand(romfile, ex=False):
+	rom = Rom()
```
Comment (mrtenda) [inline coilsnake/ui/common.py]: you're still indented with tabs here for some reason

File: coilsnake/ui/gui_util.py [hunk 4]
```diff
@@ -50,30 +60,36 @@ def add_header_to_rom(root):
...
+                parent=root,
+                title="Header Addition Failed",
+                message="Invalid ROM.")
 
 def strip_header_from_rom(root):
     filename = tkFileDialog.askopenfilename(
         parent=root,
         initialdir=os.path.expanduser("~"),
-        title="Select a ROM from which to remove a header",
+        title="Select a ROM to which to add a header",
```
Comment (mrtenda) [inline coilsnake/ui/gui_util.py]: shouldn't this text be about removing a header?

File: coilsnake/ui/cli.py [hunk 5]
```diff
@@ -72,14 +89,46 @@ def _upgrade(args):
...
+        header = True
+    else:
+        header = False
+    patch_rom(clean_rom_filename=args.clean_rom,
+              patched_rom_filename=args.output_rom,
+              patch_filename=args.patch,
+              headered=header)
+
+def _expand(args):
+    if args.exhi == "true":
```
Comment (mrtenda) [inline coilsnake/ui/cli.py]: exval = (args.exhi == "true")

File: coilsnake/ui/cli.py [hunk 6]
```diff
@@ -72,14 +89,46 @@ def _upgrade(args):
...
-					 project_path=args.project_directory)
+    decompile_script(rom_filename=args.rom_filename, 
+                     project_path=args.project_directory)
 
 def _patchrom(args):
-	patch_rom(clean_rom_filename=args.clean_rom,
-			  patched_rom_filename=args.output_rom,
-			  patch_filename=args.patch,
-			  headered=args.headered)
+    if args.headered == "true":
```
Comment (mrtenda) [inline coilsnake/ui/cli.py]: header r= args.headered == "true"

Comment (LittleCube13) [inline coilsnake/ui/cli.py]: i'm confused as to what you are telling me to do. I tried `header = args.headered == "true"` and i tried `header == args.headered == "true"` and both yielded an error. but i tested the patchrom command with an ips patch both with the false argument and the true argument, and tested the GUI the same way, and they both did the exact same thing. could you please clarify what you're trying to do?

File: coilsnake/ui/cli.py (hunk 5 above)
Comment (LittleCube13) [inline coilsnake/ui/cli.py]: i am also confused here. i combined the two expand commands into one like you said, and tested it, and it works exactly as it should. using `exval = (args.exhi == "true")` gives an incorrect syntax error, and `exval == (args.exhi == "true")` says that you're referencing a variable that doesn't exist yet. again, could you tell me what problem you're seeing with the code as is?

**Pull request merged**